eventsources/sources/kafka/start.go

```diff
@@ -223,6 +223,15 @@ func (el *EventListener) partitionConsumer(ctx context.Context, log *zap.Sugared
 			Timestamp: msg.Timestamp.String(),
 			Metadata:  kafkaEventSource.Metadata,
 		}
+
+		headers := make(map[string]string)
+
+		for _, recordHeader := range msg.Headers {
+			headers[string(recordHeader.Key)] = string(recordHeader.Value)
+		}
+
+		eventData.Headers = headers
```

Comment:
Thanks, could you also add it to `consumerGroupConsumer()`?

Reply:
That's should do it !